dev/tests/js/jasmine/tests/app/code/Magento/AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/overlay.test.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'jquery',
+    'Magento_AdobeStockImageAdminUi/js/components/grid/column/overlay'
+], function ($, Overlay) {
+    'use strict';
+
+    describe('Magento_AdobeStockImageAdminUi/js/components/grid/column/overlay', function () {
+
+        var overlayObj;
+
+        beforeEach(function () {
+            overlayObj = new Overlay({
+                provider: 'providerName'
+            });
+        });
+
+        describe('"initObservable" method', function () {
+
+            it('Check for defined ', function () {
+                expect(overlayObj.hasOwnProperty('initObservable')).toBeDefined();
+            });
+
+            it('Check method type', function () {
+                var type = typeof overlayObj.initObservable;
+
+                expect(type).toEqual('function');
+            });
+
+            it('licensed will be defined', function () {
+                expect(overlayObj.licensed).toBeDefined();
+            });
+        });
+
+        describe('"updateLicensed" method', function () {
+            it('Check for defined', function () {
+                expect(overlayObj.hasOwnProperty('updateLicensed')).toBeDefined();
+            });
+
+            it('Check method type', function () {
+                var type = typeof overlayObj.updateLicensed;
+                expect(type).toEqual('function');
+            });
+
+            it("Check Ajax Success request", function() {
+                var login = {
+                    user: function () {
+                        return {
+                            isAuthorized: true
+                        };
+                    }
+                }, ids = [1,2,3];
+                spyOn(overlayObj, 'login').and.callFake(function(e) {
+                    return login;
+                });
+                spyOn(overlayObj, 'getIds').and.callFake(function(e) {
+                    return ids;
+                });
+                spyOn($, 'ajax').and.callFake(function(e) {
+                    return $.Deferred().resolve({
+                        'response': 'success'
+                    }).promise();
+                });
+                overlayObj.updateLicensed();
+                expect(overlayObj.updateLicensed()).toBeUndefined();
+                expect($.ajax).toHaveBeenCalled();
+                expect(overlayObj.getIds).toHaveBeenCalled();
+            });
+
+            it('Check Ajax failure request', function() {
+                var login = {
+                    user: function () {
+                        return {
+                            isAuthorized: false
```

Comment:
Stubbing `isAuthorized` out to return `false` doesn't actually execute the AJAX failure flow - rather, it would execute the 'not-logged-in-so-won't-issue-an-AJAX-request' flow. So I would suggest either renaming and restructuring this test so that it reflects this, or re-write the test to actually exercise the 'AJAX request has failed' flow.

Friendly reminder that the `updateLicensed` method has three main logic flows:

1. User is not authorized, so no AJAX request is issued, the `licensed` property is set to an empy object `{}` and the method `return`s early.
2. User is authorized, so we issue the AJAX request and after receiving a successful response from the server, we set the `licensed` property to the server's response.
3. User is authorized, so we issue the AJAX request and after receiving an error response from the server or  the AJAX request  fails, we `return` the error message.

As such, to completely cover the logic of the `updateLicensed` method with unit tests, we should create three tests for this method, one to cover each main logical flow.